### PR TITLE
Using new serialization format.

### PR DESCRIPTION
**Pull request by sebastianklose**
Updates everyone to use the new serialization format.

A corresponding PR will be made in cartographer_ros for the various tools.

### CODE REVIEW

File: cartographer/cloud/client/map_builder_stub.cc [hunk 1]
```diff
@@ -77,6 +78,7 @@ int MapBuilderStub::AddTrajectoryForDeserialization(
     const mapping::proto::TrajectoryBuilderOptionsWithSensorIds&
         options_with_sensor_ids_proto) {
   LOG(FATAL) << "Not implemented";
+  return -1;
```
Comment (cschuet) [inline cartographer/cloud/client/map_builder_stub.cc]: This should compile fine w/o the return statement.

File: cartographer/ground_truth/compute_relations_metrics_main.cc [hunk 2]
```diff
@@ -126,7 +127,7 @@ void WriteRelationMetricsToFile(const std::vector<Error>& errors,
          "expected_translation_x,expected_translation_y,expected_"
          "translation_z,expected_rotation_w,expected_rotation_x,"
          "expected_rotation_y,expected_rotation_z,covered_distance\n";
-  for (size_t relation_index = 0; relation_index < ground_truth.relation_size();
+  for (int relation_index = 0; relation_index < ground_truth.relation_size();
```
Comment (cschuet) [inline cartographer/ground_truth/compute_relations_metrics_main.cc]: Why int?

File: cartographer/cloud/client/map_builder_stub.cc (hunk 1 above)
Comment (sebastianklose) [reply cartographer/cloud/client/map_builder_stub.cc]: I did get a compilation error at some point, that's why I added it.

File: cartographer/ground_truth/compute_relations_metrics_main.cc (hunk 2 above)
Comment (sebastianklose) [reply cartographer/ground_truth/compute_relations_metrics_main.cc]: compiler warning about comparison between signed and unsigned. `relation_size()` is int.

File: cartographer/cloud/client/map_builder_stub.cc (hunk 1 above)
Comment (sebastianklose) [reply cartographer/cloud/client/map_builder_stub.cc]: removed.

**Comment by pifon2a** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user pifon2a. Merge queue now has a length of 1.

**Pull request merged**